### ISSUE TITLE
Support `Number.NaN` within `use-isnan`

### ISSUE THREAD

**Issue by esdmr** [eslint/eslint#14715]
**What rule do you want to change?**
`use-isnan`

**Does this change cause the rule to produce more or fewer warnings?**
More.

**How will the change be implemented? (New option, new default behavior, etc.)?**
Perhaps as a default behaviour, as `NaN` and `Number.NaN` are equivalent.

**Please provide some example code that this change will affect:**

```js
x === NaN;
x === Number.NaN;
```

**What does the rule currently do for this code?**
Only the first line is reported as a NaN comparison. Second line does not report.

**What will the rule do after it's changed?**
Both lines will report as NaN comparisons. (Maybe recommending `Number.isNaN` instead of `isNaN` but that is aside)

**Are you willing to submit a pull request to implement this change?**
I would prefer if someone else send a PR.

At [`lib/rules/use-isnan.js:24`](https://github.com/eslint/eslint/blob/353ddf9/lib/rules/use-isnan.js#L24):

```diff
-    return Boolean(node) && node.type === "Identifier" && node.name === "NaN";
+    return Boolean(node) && (node.type === "Identifier" && node.name === "NaN" ||
+        node.type === "MemberExpression" &&
+        node.object.type === "Identifier" && node.object.name === "Number" &&
+        node.property.type === "Identifier" && node.property.name === "NaN");
```

**Comment by snitin315** [comment]
Looks like a bug to me. 

> I would prefer if someone else send a PR.

I can send a PR if this issue is accepted.

**Comment by aladdin-add** [comment]
agreed it was a bug.

**Comment by nzakas** [comment]
Agree, looks like a bug.
